docs/src/main/sphinx/glossary.rst

```diff
@@ -0,0 +1,204 @@
+
+.. _glossJKS:
+
+Java KeyStore (JKS)
+    The system of public key cryptography supported as one part of the Java
+    security APIs. The legacy JKS system recognizes keys and certificates stored
+    in *keystore* files, typically with the ``.jks`` extension, and relies on a
+    system-level list of CAs in *truststore* files installed as part of the
+    current Java installation.
+
+Key
+    A cryptographic key specified as a pair of public and private strings
+    generally used in the context of :ref:`TLS <glossTLS>` to secure public
+    network traffic.
+
+.. _glossLB:
+
+Load Balancer (LB)
+    Software or a hardware device that sits on a network edge and accepts
+    network connections on behalf of servers behind that wall, distributing
+    traffic across network and server infrastructure to balance the load on
+    networked services.
+
+.. _glossObjectStorage:
+
+Object Storage
+    `Object storage <https://en.wikipedia.org/wiki/Object_storage>`_ is a file
+    storage mechanism that stores data in a randomly-accessed manner, as opposed
+    to hierarchical filesystems or sector-based block storage. In the context of
+    Trino, "object storage" commonly refers to
+    `cloud storage <https://en.wikipedia.org/wiki/Object_storage#Cloud_storage>`_
+    technologies such as `Amazon S3 <https://aws.amazon.com/s3>`_,
+    `Google Cloud Storage <https://cloud.google.com/storage>`_, and
+    `Azure Blob Storage <https://azure.microsoft.com/en-us/products/storage/blobs>`_.
+    In addition to cloud-hosted services, there are also local object storage
+    options such as `MinIO <https://min.io/>`_ and
+    `Ceph <https://docs.ceph.com>`_ that are compatible with S3. Object storage
+    became a popular replacement to :ref:`HDFS <glossHDFS>`.
+
+.. _glossOpenSource:
+
+Open-source
+    Typically refers to
+    `open-source software <https://en.wikipedia.org/wiki/Open-source_software>`_.
+    which is software that has the source code made available for others to see,
+    use, and contribute to. Allowed usage varies depending on the license that
+    the software is licensed under. Trino is licensed under the
+    `Apache license <https://en.wikipedia.org/wiki/Apache_License>`_, and is
+    therefore maintained by a community of contributors from all across the
+    globe.
+
+.. _glossPlugin:
+
+Plugin
+    A bundle of code implementing the Trino
+    :doc:`Service Provider Interface (SPI) </develop/spi-overview>` that is used
+    to add new :ref:`connectors <glossConnector>`,
+    :doc:`data types </develop/types>`, :doc:`functions`,
+    :doc:`access control implementations </develop/system-access-control>`, and
+    other features of Trino.
+
+Plugin
+    Any code implementing the :doc:`SPI </develop/spi-overview>` used to add
+    new :ref:`connectors <glossConnector>`, :doc:`data types </develop/types>`,
+    :doc:`functions`, :doc:`/develop/system-access-control`, and other features of Trino.
+
+Privacy-Enhanced Mail (PEM)
```

Comment:
"PEM file format" is probably a better title. The original name is historical artifact that isn't related to its current usage. I didn't even know what it meant and was confused why we had "mail" in the glossary.

Reply:
We were trying to stick to a format of <name> (<acronym>) but I think this is better to avoid adding more concepts to the reader.